### PR TITLE
Add a console command for removing expired tokens

### LINKED ISSUE #4: Add console commands for stale token cleanup

### PR DESCRIPTION
**Pull request by HypeMC**
Closes #4

### CODE REVIEW

File: Command/ClearExpiredTokensCommand.php [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+    protected function execute(InputInterface $input, OutputInterface $output): int
+    {
+        $io = new SymfonyStyle($input, $output);
+
+        $clearExpiredAccessTokens = !$input->getOption('refresh-tokens-only');
+        $clearExpiredRefreshTokens = !$input->getOption('access-tokens-only');
+
+        if (!$clearExpiredAccessTokens && !$clearExpiredRefreshTokens) {
+            $io->error('Please chose only one of the following options: "access-tokens-only", "refresh-tokens-only".');
```
Comment (X-Coder264) [inline Command/ClearExpiredTokensCommand.php]: @HypeMC Chose -> choose typo :)

File: Manager/Doctrine/AccessTokenManager.php [hunk 2]
```diff
@@ -34,4 +34,13 @@ public function save(AccessToken $accessToken): void
         $this->entityManager->persist($accessToken);
         $this->entityManager->flush();
     }
+
+    public function clearExpired(): int
+    {
+        return $this->entityManager->createQueryBuilder()
+            ->delete(AccessToken::class, 'at')
+            ->where('at.expiry < CURRENT_TIMESTAMP()')
```
Comment (X-Coder264) [inline Manager/Doctrine/AccessTokenManager.php]: @HypeMC I'm not sure how I feel about the usage of `CURRENT_TIMESTAMP()` here. Using it means that this feature won't work on all databases supported by Doctrine. On the other hand binding an object (DateTime in this case) via the `setParameter()` method is "broken" (https://github.com/doctrine/orm/pull/7528) in Doctrine 2.6.3 (in our internal application that uses this bundle) and 2.64 which has the fix for it hasn't been tagged yet. What is your opinion on this matter?

File: Tests/Acceptance/ClearExpiredTokensCommandTest.php [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+        $commandTester = new CommandTester($command);
+
+        $exitCode = $commandTester->execute([
+            'command' => $command->getName(),
+        ]);
+
+        $this->assertSame(0, $exitCode);
+
+        $output = $commandTester->getDisplay();
+        $this->assertStringContainsString('Cleared 1 expired access token.', $output);
```
Comment (X-Coder264) [inline Tests/Acceptance/ClearExpiredTokensCommandTest.php]: @HypeMC These and some other similar tests in this PR don't actually test the feature in question. They only test the string output of the command which makes them kinda useless in its current state. The test should have assertions that prove that the tokens get actually cleared properly in the database.

File: Tests/Unit/InMemoryAccessTokenManagerTest.php [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+        timecop_freeze(new DateTime());
+
+        $testData = $this->buildClearExpiredTestData();
+
+        foreach ($testData['input'] as $token) {
+            $inMemoryAccessTokenManager->save($token);
+        }
+        $this->assertSame(3, $inMemoryAccessTokenManager->clearExpired());
+
+        timecop_return();
```
Comment (X-Coder264) [inline Tests/Unit/InMemoryAccessTokenManagerTest.php]: @HypeMC Please use the setUp/tearDown methods for (un)freezing time or wrap this chunk of the code in a try {} finally {} block. The way the test is currently written if the assertion above the `timecop_return` call fails an exception will be thrown which means that `timecop_return` won't be called and all the rest of the tests will be run with frozen time. The same applies for the `InMemoryRefreshTokenManagerTest` too.

File: docs/basic-setup.md [hunk 5]
```diff
@@ -153,6 +153,24 @@ There are two possible reasons for the authentication server to reject a request
 - Provided token is expired or invalid (HTTP response 401 `Unauthorized`)
 - Provided token is valid but scopes are insufficient (HTTP response 403 `Forbidden`)
 
+## Clearing expired access & refresh tokens
+
+To clear expired access & refresh tokens you can use the `trikoder:oauth2:clear-expired-tokens` command.
+
+The command removes all tokes whose expiry time is lesser than the current.
```
Comment (X-Coder264) [inline docs/basic-setup.md]: @HypeMC Tokes -> tokens typo :)

File: Manager/Doctrine/AccessTokenManager.php (hunk 2 above)
Comment (X-Coder264) [reply Manager/Doctrine/AccessTokenManager.php]: nvm, CURRENT_TIMESTAMP() is actually a DQL function which gets parsed properly.

File: Command/ClearExpiredTokensCommand.php (hunk 1 above)
Comment (HypeMC) [reply Command/ClearExpiredTokensCommand.php]: Done

File: docs/basic-setup.md (hunk 5 above)
Comment (HypeMC) [reply docs/basic-setup.md]: Done

File: Tests/Acceptance/ClearExpiredTokensCommandTest.php (hunk 3 above)
Comment (HypeMC) [reply Tests/Acceptance/ClearExpiredTokensCommandTest.php]: @X-Coder264 Added two additional tests, currently in the acceptance tests folder, hopefully will be moved at a later time once the infrastructure is improved.

File: Manager/Doctrine/AccessTokenManager.php (hunk 2 above)
Comment (HypeMC) [reply Manager/Doctrine/AccessTokenManager.php]: @X-Coder264 Switched to DateTime after all for consistency & cause timecop doesn't work with `CURRENT_TIMESTAMP`.

File: Tests/Unit/InMemoryAccessTokenManagerTest.php (hunk 4 above)
Comment (HypeMC) [reply Tests/Unit/InMemoryAccessTokenManagerTest.php]: @X-Coder264 Agreed, there are several tests that use timecop incorrectly, I'll create a separate PR to fix all of them at once, so I'd leave this as is for now.

**Pull request merged**